packages/experiments/src/Foundation.ts

```diff
@@ -78,17 +73,34 @@ export function createComponent<TComponentProps, TViewProps, TStyleSet extends I
     TViewProps,
     TStyleSet,
     IProcessedStyleSet<TStyleSet>,
-    IContextCustomization,
+    ICustomizerContext,
     ITheme,
+    IScheme,
     TStatics
   >(options, providers);
 }
 
-// TODO: remove any if possible
-// tslint:disable-next-line:no-any
-function getCustomizations(displayName: string, context: IContextCustomization): any {
+function getCustomizations<TViewProps, TStyleSet>(
+  displayName: string,
+  context: ICustomizerContext
+): IStyleableComponentProps<TViewProps, TStyleSet, ITheme, IScheme> {
   // TODO: do we want field props? should fields be part of IComponentOptions and used here?
   // TODO: should we centrally define DefaultFields? (not exported from styling)
   const DefaultFields = ['theme', 'styles'];
   return Customizations.getSettings(DefaultFields, displayName, context.customizations);
 }
+
+function getContextFromProps<TViewProps, TStyleSet>(
+  props: IStyleableComponentProps<TViewProps, TStyleSet, ITheme, IScheme>,
```

Comment:
I believe the way you've set it up, this method gets called from `getCustomizations` on *every render, causing us to recalc the schemed context every time. Can you optimize the render to avoid anything unnecessary? We shouldn't have to recalc a scheme if it never changed.

Reply:
This will be tricky because we can't assume scheme data hasn't changed even if scheme name is the same. mergeCustomizations makes new objects, as well, which means shallow refs won't work. I'll take with a TODO for now.